python/lsst/sims/catalogs/generation/db/ObservationMetaData.py

```diff
@@ -60,70 +60,94 @@ class ObservationMetaData(object):
-                 bandpassName='r', phoSimMetadata=None, site=None, m5=None, skyBrightness=None):
+                 mjd=None, unrefractedRA=None, unrefractedDec=None, rotSkyPos=None,
+                 bandpassName=None, phoSimMetaData=None, site=Site(), m5=None, skyBrightness=None):
 
-        self.bounds = None
-        self.boundType = boundType
-        self.mjd = mjd
-        self.bandpass = bandpassName
-        self.skyBrightness = skyBrightness
+        self._bounds = None
+        self._boundType = boundType
+        self._mjd = mjd
+        self._bandpass = bandpassName
+        self._skyBrightness = skyBrightness
+        self._site = site
 
         if rotSkyPos is not None:
-            self.rotSkyPos = numpy.radians(rotSkyPos)
+            self._rotSkyPos = numpy.radians(rotSkyPos)
         else:
-            self.rotSkyPos = None
+            self._rotSkyPos = None
 
         if unrefractedRA is not None:
-            self.unrefractedRA = numpy.radians(unrefractedRA)
+            self._unrefractedRA = numpy.radians(unrefractedRA)
         else:
-            self.unrefractedRA = None
+            self._unrefractedRA = None
 
         if unrefractedDec is not None:
-            self.unrefractedDec = numpy.radians(unrefractedDec)
+            self._unrefractedDec = numpy.radians(unrefractedDec)
         else:
-            self.unrefractedDec = None
+            self._unrefractedDec = None
 
         if boundLength is not None:
-            if isinstance(boundLength, float):
-                self.boundLength = numpy.radians(boundLength)
-            else:
-                try:
-                    self.boundLength = []
-                    for ll in boundLength:
-                        self.boundLength.append(numpy.radians(ll))
-                except:
-                    raise RuntimeError("You seem to have passed something that is neither a float nor " +
-                                       "list-like as boundLength to ObservationMetaData")
+            self._boundLength = numpy.radians(boundLength)
         else:
-            self.boundLength = None
+            self._boundLength = None
 
-        if site is not None:
-            self.site=site
+        if phoSimMetaData is not None:
+            self._assignPhoSimMetaData(phoSimMetaData)
         else:
-            self.site=Site()
+            self._phoSimMetaData = None
 
-        if site is not None:
-            self.site=site
-        else:
-            self.site=Site()
+        self._assignM5(m5)
 
-        if phoSimMetadata is not None:
-            self.assignPhoSimMetaData(phoSimMetadata)
-        else:
-            self.phoSimMetadata = None
+        #this should be done after phoSimMetaData is assigned, just in case
+        #self._assignPhoSimMetadata overwrites unrefractedRA/Dec
+        if self._bounds is None:
+            self._buildBounds()
+
+
+    @property
+    def summary(self):
```

Comment:
Yeah.  You have a better sense of how this should be used.  Make it a separate ticket.